client/blocks/app-banner/utils.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Internal dependencies
+ */
+import {
+	get,
+	includes,
+	reduce,
+} from 'lodash';
+
+export const PREFERENCE_NAME = 'appBannerDismissTimes';
+export const EDITOR = 'post-editor';
+export const NOTES = 'notifications';
+export const READER = 'reader';
+export const STATS = 'stats';
+export const ALLOWED_SECTIONS = [ EDITOR, NOTES, READER, STATS ];
+export const ONE_WEEK_IN_MILLISECONDS = 604800000;
+export const ONE_YEAR_IN_MILLISECONDS = 31540000000;
+
+export function getAppBannerData( translate, sectionName ) {
+	switch ( sectionName ) {
+		case EDITOR:
+			return {
+				title: translate( 'Rich mobile publishing.' ),
+				copy: translate( 'A streamlined editor with faster, simpler image uploading? Check and mate.' ),
+			};
+		case NOTES:
+			return {
+				title: translate( 'Watch engagement happening.' ),
+				copy: translate( 'Is your new post a hit? With push notifications, see reactions as they roll in.' ),
+			};
+		case READER:
+			return {
+				title: translate( 'Read online or off.' ),
+				copy: translate( 'Catch up with new posts when you have time, even if you are offline.' ),
+			};
+		case STATS:
+			return {
+				title: translate( 'Stats at your fingertips.' ),
+				copy: translate( "Add real-time stats to your device's notifications or widgets." ),
+			};
+		default:
+			return {
+				title: '',
+				copy: '',
+			};
+	}
+}
+
+export function getCurrentSection( currentSection, isNotesOpen ) {
+	if ( isNotesOpen ) {
+		return NOTES;
+	}
+
+	if ( includes( ALLOWED_SECTIONS, currentSection ) ) {
+		return currentSection;
+	}
+
+	return null;
+}
+
+export function getNewDismissTimes( dismissedSection ) {
+	const currentTime = Date.now();
+	const aWeekFromNow = currentTime + ONE_WEEK_IN_MILLISECONDS;
+	const aYearFromNow = currentTime + ONE_YEAR_IN_MILLISECONDS;
+
+	return reduce( ALLOWED_SECTIONS, ( result, section ) => {
+		result[ section ] = ( section === dismissedSection ) ? aYearFromNow : aWeekFromNow;
```

Comment:
I thought we settled on option 1 from https://github.com/Automattic/wp-calypso/issues/15228:
`Option 1: dismissing banner x turns off banner x for one year, also turns off all other banners for one week. After one week, banner x is still off, but user would see banner y in context`.

I think with this setup, it would be very unlikely to be greeted every Monday morning at tea time :smile:, because the dismissed sections won't be shown again for one year. So I'm guessing the most likely scenario would be to first notice it in Stats/Reader, dismiss it, and a week after that in Notifications (probably won't be opened at the same time).

Having said that, I agree that one week seems frequent, and we could maybe bump that to a month. In either case, I don't think there is much need for randomization, but I don't mind adding it. @gwwar what do you think? 

Also, I think this discussion is better suited for https://github.com/Automattic/wp-calypso/issues/15228 so we can loop in other involved parties.